### ISSUE TITLE
os_network missing --dns-domain / dns based on hostnames not working in tenant

### ISSUE THREAD

**Issue by Thingee** [ansible/ansible#19622]
_From @geektoor on November 28, 2016 19:52_

##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
os_network or os_subnet module

##### ANSIBLE VERSION

```
ansible 2.1.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

default settings

##### OS / ENVIRONMENT

Ubuntu Mate 16.04.1 LTS

##### SUMMARY

While trying to figure out why DNS within the tenant is not working, i found the following issue:
If i use os_network to create a network, there is no option for specifying the dns-domain for that network. I also checked os_subnet but this also doesn't provide anything to add the dns domain for that network. The issue here is, that without that option, DNS resolving within the tenant is not working as expected. This is due to the fact, that without a DNS domain, the only dns-entries OpenStack is providing, are the "host-ipoct1-ipoct2-ipoct3-ipoct4" entries. No "hostname" entries at all. As soon as i issue a "neutron net-update --dns-domain bla. $NETWORK_ID" the hosts get updated and name resolving does work.

##### STEPS TO REPRODUCE

Steps to reproduce:
- create a network using ansible os_network
- add subnet to this network
- create two instances and attach it to the network
- (optional, depending on setup) attach floating IPs to the instances
- access instance one
- try to "ping instance2"
- see failure.

```
  - name: create internal network
    os_network:
      cloud: "mycloud"
      name: "internal-lan"
      state: present

  - name: Create subnet on internal network
    os_subnet:
      cloud: "mycloud"
      network_name: "internal-lan"
      name: "mysubnet"
      cidr: "192.168.100.0/24"
      allocation_pool_start: "192.168.100.50"
      allocation_pool_end: "192.168.100.150"
      state: present
```

##### EXPECTED RESULTS

I'd expect to have an option to specify a DNS domain name for my network. The "real issue" behind the scene can be that this is more an issue within OpenStack, but in the end, the simple option to add the dns-domain fixes it. So the option is missing (if thats more like a feature request, i'm fine).

##### ACTUAL RESULTS

DNS resolving within the tenant don't work. At least not based on instance names.

_Copied from original issue: ansible/ansible-modules-core#5753_

**Comment by Thingee** [comment]
_From @ansibot on November 28, 2016 20:4_


@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, @thingee, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @ansibot on December 7, 2016 20:41_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
cc @dagnello
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @omgjlk
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @mnaser
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mnaser** [comment]
needs_contributor

It would be nice to have someone add this, it should be straight-forward.

**Comment by ansibot** [comment]
cc @cloudnull @evrardjp @odyssey4me
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @gtema
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
